source/18_docker_and_containers.rst

```diff
@@ -0,0 +1,196 @@
+Docker and Containers
+=====================
+
+*New and Improved!*
+-------------------
+* Configuration management is approaching 15 years old.
+* Some of the command line utilities we use date back to times when people used
+  electric typewriters instead of computer screens (we aren't kidding)
+* Today we're going to explore a technology which is in its angsty teenage
+  years, it's not fully mature.
+
+
+Quick Review of VMs
+-------------------
+* The host OS is the one running on bare metal, the OS which the VM runs
+  inside.
+* The guest OS is the OS running inside the VM.
+* Software running inside VMs is slower than software running on the host
+  machine.
+
+.. figure:: /static/nesting_dolls.png
+    :align: center
+    :scale: 70%
+
+
+What are Containers?
+--------------------
+
+* Not VMs.
+* Containers are a way to put a program in an imaginary box where it thinks
+  it's the only program besides the OS running on the computer.
+* Containers are just the host OS lying to the program, so they don't need to
+  run a second OS.
+* Containers allow programs to be isolated from the host system without the
+  overhead of VMs. Since VMs run a whole OS on top of an OS, they're slow.
+* As a side affect, it changes the way you configure and run your applications.
+
+A Brief history of Containers
+-----------------------------
+
+* The ``chroot`` command is a way to change the location of the root directory
+  for a process. Using ``chroot`` you can effectively change the root directory
+  for your entire OS so processes running inside the ``chroot`` can't change
+  files outside the ``chroot``. ``chroot`` was introduced in Unix in 1979.
+* FreeBSD, an OS very similar to Linux, has had container like things since the
+  2000s, but they never saw widespread use.
```

Comment:
There is also OpenVZ, which is more containerization than proper virtualization.